modules/distribution/resources/customsequences/fault/json_fault.xml

```diff
@@ -0,0 +1,41 @@
+<sequence xmlns="http://ws.apache.org/ns/synapse" name="json_fault">
+    <log level="custom">
+        <property name="STATUS" value="Executing custom 'fault' sequence"/>
+        <property name="ERROR_CODE" expression="get-property('ERROR_CODE')"/>
+        <property name="ERROR_MESSAGE" expression="get-property('ERROR_MESSAGE')"/>
+    </log>
+    <payloadFactory>
+        <format>
+            <am:fault xmlns:am="http://wso2.org/apimanager">
+                <am:code>$1</am:code>
+                <am:type>Status report</am:type>
+                <am:message>Runtime Error</am:message>
+                <am:description>$2</am:description>
+            </am:fault>
+        </format>
+        <args>
+            <arg expression="$ctx:ERROR_CODE"/>
+            <arg expression="$ctx:ERROR_MESSAGE"/>
+        </args>
+    </payloadFactory>
+    <filter xpath="$ctx:CUSTOM_HTTP_SC">
+        <then>
+            <property name="HTTP_SC" expression="$ctx:CUSTOM_HTTP_SC" scope="axis2"/>
+        </then>
+        <else>
+            <property name="HTTP_SC" value="500" scope="axis2"/>
+        </else>
+    </filter>
+    <class name="org.wso2.carbon.apimgt.usage.publisher.APIMgtFaultHandler"/>
+    <property name="RESPONSE" value="true"/>
```

Comment:
you can use <respond/> mediator in place of following 4 mediators
<property name="RESPONSE" value="true"/>
<header name="To" action="remove"/>
<send/>
<drop/>